src/stylize.ts

```diff
@@ -97,3 +101,30 @@ export function nominalColorScaleForHighCardinality(specM: SpecQueryModel, schem
 
   return specM;
 }
+
+export function xAxisOnTopForHighYCardinalityWithoutColumn(specM: SpecQueryModel, schema: Schema): SpecQueryModel {
+ [Channel.COLUMN, Channel.X, Channel.Y].forEach((channel) => {
+        encQIndex[channel] = specM.getEncodingQueryByChannel(channel);
```

Comment:
I just realize that you choose a terrible location to declare `encQIndex`, which can make all of these functions incorrect when you run any of these functions for the second time!